admin/server/auth/middleware.go

```diff
@@ -90,6 +90,21 @@ func (a *Authenticator) HTTPMiddlewareLenient(next http.Handler) http.Handler {
 	return a.httpMiddleware(next, true)
 }
 
+// HTTPMiddlewareWithCookieRefresh is a middleware that refreshes the auth cookie
+func (a *Authenticator) HTTPMiddlewareWithCookieRefresh(next http.Handler) http.Handler {
```

Comment:
1. Can you name this `CookieRefreshHTTPMiddleware`? It sounds like this is similar to the `HTTPMiddleware` above, but that is not the case – this middleware does something different.
2. Maybe add a bit extra info in the docstring, e.g.:
```suggestion
// CookieRefreshMiddleware is a middleware that refreshes the auth cookie.
// This enables us to do rolling cookie refreshes so we can have a relatively short cookie max age.
// Note that it does not update the auth token encrypted inside the cookie.
func (a *Authenticator) CookieRefreshMiddleware(next http.Handler) http.Handler {
```